From 5f673ed1b88756347205efcc0596f63d42d0df3a Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 22 Sep 2016 13:32:07 -0500
Subject: [PATCH 14/35] Revert "drm/tilcdc: Remove unnecessary pm_runtime_get()
 and *_put() calls"

This reverts commit 76fb2b46e6ea135b9bc0a3cf643c5c7ed3a25a85.
---
 drivers/gpu/drm/tilcdc/tilcdc_crtc.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/gpu/drm/tilcdc/tilcdc_crtc.c b/drivers/gpu/drm/tilcdc/tilcdc_crtc.c
index 3b3fed5..c2dfb4b 100644
--- a/drivers/gpu/drm/tilcdc/tilcdc_crtc.c
+++ b/drivers/gpu/drm/tilcdc/tilcdc_crtc.c
@@ -243,6 +243,8 @@ int tilcdc_crtc_page_flip(struct drm_crtc *crtc,
 
 	crtc->primary->fb = fb;
 
+	pm_runtime_get_sync(dev->dev);
+
 	spin_lock_irqsave(&tilcdc_crtc->irq_lock, flags);
 
 	if (crtc->hwmode.vrefresh && ktime_to_ns(tilcdc_crtc->last_vblank)) {
@@ -265,6 +267,8 @@ int tilcdc_crtc_page_flip(struct drm_crtc *crtc,
 
 	spin_unlock_irqrestore(&tilcdc_crtc->irq_lock, flags);
 
+	pm_runtime_put_sync(dev->dev);
+
 	return 0;
 }
 
@@ -313,6 +317,8 @@ static void tilcdc_crtc_mode_set_nofb(struct drm_crtc *crtc)
 	if (WARN_ON(!fb))
 		return;
 
+	pm_runtime_get_sync(dev->dev);
+
 	/* Configure the Burst Size and fifo threshold of DMA: */
 	reg = tilcdc_read(dev, LCDC_DMA_CTRL_REG) & ~0x00000770;
 	switch (info->dma_burst_sz) {
@@ -461,6 +467,8 @@ static void tilcdc_crtc_mode_set_nofb(struct drm_crtc *crtc)
 
 	tilcdc_crtc_update_clk(crtc);
 
+	pm_runtime_put_sync(dev->dev);
+
 	crtc->hwmode = crtc->state->adjusted_mode;
 }
 
-- 
2.9.3

